pkg/agent/datamodel/helper.go

```diff
@@ -93,3 +93,24 @@ func IndentString(original string, spaces int) string {
 	}
 	return out.String()
 }
+
+func trimEOF(data []byte) []byte {
+	eofIndex := bytes.LastIndex(data, []byte("#EOF"))
+	if eofIndex != -1 { // #EOF found
+		newlineIndex := bytes.LastIndex(data[:eofIndex], []byte("\n"))
+		if newlineIndex != -1 {
+			return data[:newlineIndex]
+		}
+	}
+	return data
+}
+
+func processDownloadURL(downloadURL string) string {
+	// example URL "downloadURL": "mcr.microsoft.com/oss/kubernetes/autoscaler/addon-resizer:*",
+	// getting the data between the last / and the last :
+	parts := strings.Split(downloadURL, "/")
```

Comment:
we should probably be checking lens of the slices return from the `Split` calls to ensure that the URL is in the correct format while processing

Reply:
Actually in the function I do 

lastPart := parts[len(parts)-1]
component := strings.Split(lastPart, ":")

so that I'm always looking at the last / to the final : "/addon-resizer:"

so instead of a / check, I think ensuring that component[0] exists is what will be important